pkg/apis/scheduling/v1alpha1/types_locationdomain.go

```diff
@@ -0,0 +1,175 @@
+}
+
+// InstancesReference describes a reference to a workspace holding the instances
+// subject to the location domain. Exactly one of the fields must be set.
+type InstancesReference struct {
+	// resource is the group version resource of the instances that are subject to this location
+	// domain. The creator of this location domain needs to have verb "create" permission
+	// in the referenced workspace on the given resource with the subresource "locationdomain".
+	//
+	// +required
+	// +kubebuilder:Required
+	Resource GroupVersionResource `json:"resource"`
+
+	// workspace is a reference to a workspace with instances subject to the location
+	// domain. Exactly one must be set.
+	//
+	// +optional
+	Workspace *WorkspaceExportReference `json:"workspace,omitempty"`
+}
+
+// GroupVersionResource unambiguously identifies a resource.
+type GroupVersionResource struct {
+	// group is the name of an API group.
+	//
+	// +kubebuilder:validation:Pattern=`^(|[a-z0-9]([-a-z0-9]*[a-z0-9](\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*)?)$`
+	// +optional
+	Group string `json:"group,omitempty"`
+
+	// rersion is the version of the API.
+	//
+	// +kubebuilder:validation:Pattern=`^[a-z]([-a-z0-9]*[a-z0-9]$`
+	// +kubebuilder:validation:MinLength:1
+	// +required
+	// +kubebuilder:Required
+	Version string `json:"version"`
+
+	// resource is the name of the resource.
+	// +kubebuilder:validation:Pattern=`^[a-z]([-a-z0-9]*[a-z0-9]$`
+	// +kubebuilder:validation:MinLength:1
+	// +required
+	// +kubebuilder:Required
+	Resource string `json:"resource"`
+}
+
+// WorkspaceExportReference describes an API and backing implementation that are provided by an actor in the
+// specified Workspace.
+type WorkspaceExportReference struct {
+	// name is a workspace name meaning a workspace defined in the workspace this LocationDomain is defined it.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	WorkspaceName WorkspaceName `json:"name"`
+}
+
+// WorkspaceName is a non-absolute workspace name, e.g. "workspace", but not "root:company:workspace".
+//
+// +kube:validation:MinLength=1
+// +kubebuilder:validation:Pattern=`^[a-z][a-z0-9-]*[a-z0-9]$`
+type WorkspaceName string
+
+// WorkspaceSelector selects a subset of workspaces this location domain applies to by default.
+type WorkspaceSelector struct {
+	// labels are used to filter workspaces.
+	//
+	// +optional
+	Labels metav1.LabelSelector `json:"labels,omitempty"`
+
+	// priority is used when multiple LocationDomains and their workspace selector match. The largest number wins.
+	Priority uint32 `json:"priority,omitempty"`
+}
+
+type LocationDomainLocationDefinition struct {
+	LocationSpecBase `json:",inline"`
+
+	// labelSelector chooses the instances that will be part of this location.
+	//
+	// Note that these labels are not what is shown in the Location objects to
+	// the user. Depending on context, both will match or won't match.
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
```

Comment:
this is to select instances for this location, while the AvailableSelectorLabel is a constraint on whether certain labels on the instance can be used to schedule the workload to this instance. Is it correct?

Reply:
LabelSelector selects the WorkloadCluster. Labels will become labels on the Locations. `AvailableSelectorLabels` are further labels that can be used to select nodes (and potentially also locations, if there is no capacity of nodes with those labels in a workload cluser).